experimental/dds/tree2/src/feature-libraries/editable-tree-2/proxies/types.ts

```diff
@@ -235,12 +242,25 @@ export type ProxyNodeUnion<TTypes extends AllowedTypes> = TTypes extends readonl
  * Given a node's schema, return the corresponding object in the proxy-based API.
  * @alpha
  */
-export type ProxyNode<TSchema extends TreeSchema> = TSchema extends LeafSchema
+export type ProxyNode<
+	TSchema extends TreeSchema,
+	API extends "javaScript" | "sharedTree" = "sharedTree",
+> = TSchema extends LeafSchema
 	? SchemaAware.InternalTypes.TypedValue<TSchema["leafValue"]>
 	: TSchema extends MapSchema
-	? SharedTreeMap<TSchema>
+	? API extends "sharedTree"
+		? SharedTreeMap<TSchema>
+		: Map<string, ProxyField<TSchema["mapFields"], API>>
 	: TSchema extends FieldNodeSchema
-	? ProxyField<TSchema["structFieldsObject"][""]>
+	? ProxyField<TSchema["structFieldsObject"][""], API>
```

Comment:
FYI - When merging with the 'inifinite type recursion fix' PR, we'll probably need to inline the logic that constructs the SharedTreeList or readonly Array here.

Reply:
I went ahead and made the change in this PR so I don't forget. I also made the case in `ProxyField` that used to delegate to this one return `never`, just to be clear that we don't support it in general.